include/picolibrary/testing/unit/microchip/mcp3008.h

```diff
@@ -0,0 +1,103 @@
+/**
+ * picolibrary
+ *
+ * Copyright 2021 Andrew Countryman <apcountryman@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/**
+ * \file
+ * \brief picolibrary::Testing::Unit::Microchip::MCP3008 interface.
+ */
+
+#ifndef PICOLIBRARY_TESTING_UNIT_MICROCHIP_MCP3008_H
+#define PICOLIBRARY_TESTING_UNIT_MICROCHIP_MCP3008_H
+
+#include "gmock/gmock.h"
+#include "picolibrary/error.h"
+#include "picolibrary/microchip/mcp3008.h"
+#include "picolibrary/result.h"
+#include "picolibrary/testing/unit/spi.h"
+#include "picolibrary/utility.h"
+
+/**
+ * \brief Microchip MCP3008 unit testing facilities.
+ */
+namespace picolibrary::Testing::Unit::Microchip::MCP3008 {
+
+/**
+ * \brief Mock Microchip MCP3008 driver.
+ */
+class Mock_Driver {
+  public:
+    /**
+     * \copydoc picolibrary::Microchip::MCP3008::Driver::Controller
+     */
+    using Controller = ::picolibrary::Testing::Unit::SPI::Mock_Controller;
+
+    /**
+     * \copydoc picolibrary::Microchip::MCP3008::Driver::Device_Selector
+     */
+    using Device_Selector = ::picolibrary::Testing::Unit::SPI::Mock_Device_Selector::Handle;
```

Comment:
`::picolibrary::Testing::Unit::SPI::Mock_Device_Selector::Handle` does not need to be fully qualified.